docs/docs/basics/types.md

```diff
@@ -2,89 +2,224 @@
+    5 * 1000000000000000000 :: Integer
+    ```
+
+=== "In a file"
+
+    ```haskell
+    y1 :: Integer
+    y1 = 5
+
+    y2 :: Integer
+    y2 = 5 * 1000000000000000000
+    ```
+
+An expression like `5` that can more than one type is said to be *overloaded*. It can describe different values depending on the type it's used as.
 
 ??? Gotcha
-    `5` can have a more general type in Haskell. See [here](/faq/numbers.md)
+    If you ask for the type of `5` in the REPL, you'll see neither the type `Int` nor `Integer`, but rather `Num a => a`. This will make more sense once you understand type variables and type classes, but it just means the expression `5` can represent a value of any number type. See [here](/faq/numbers.md)
 
-## The type of real numbers
+## Types of real numbers
 
-There are several available options. A good general choice is `Double`:
+When it comes to numbers with a fractional part, Haskell again provides several possible types.
 
-```haskell
-5.0 :: Double
-```
+A good general choice is `Double`, which represents an IEEE double-precision floating point number:
+
+=== "In a repl"
 
-## The type of text
+    ```haskell
+    5 :: Double -- (1)!
+    3.14 :: Double
+    ```
 
-`Char` is the type of single characters:
+    1. The same expression, `5`, can also be a `Double`.  To be more explicit, you can also write `5.0`.
 
-```hs title="repl example"
-> :t 'a'
-'a' :: Char
+=== "In a file"
 
-> :t 'b'
-'b' :: Char
-```
+    ```haskell
+    z1 :: Double
+    z1 = 5
 
-`Text` is the type of sequences of characters:
+    z2 :: Double
+    z2 = 3.14
+    ```
 
-```haskell
-{-# LANGUAGE OverloadedStrings #-} --(1)!
-import Data.Text (Text)
+Floating point numbers are fixed precision, so doing computations with them can result in rounding error.  An exact option for numbers with a fractional part is `Rational`, which can represent any rational number (that is, any fraction) exactly.
+
+=== "In a repl"
+
+    ```haskell
+    5 :: Rational
+    3.14 :: Rational
+    ```
+
+=== "In a file"
+
+    ```haskell
+    z3 :: Rational
+    z3 = 5
 
-exampleText :: Text 
-exampleText = "hello world!"
-``` 
+    z4 :: Rational
+    z4 = 3.14
+    ```
+
+??? Gotcha
+    If you ask for the type of `3.14` in the REPL, you'll see neither `Double` nor `Rational`, but rather `Fractional a => a`. This will make more sense once you understand type variables and type classes, but it just means the expression `3.14` can represent a value of any fractional number type. See [here](/faq/numbers.md)
+
+## Text types
+
+`Char` is the type of single characters.  These are written in single quotes.
+
+=== "In a repl"
+
+    ```haskell
+    'A' :: Char
+    'b' :: Char
+    ```
+
+=== "In a file"
 
-1. See [here](/gotchas/strings) for why this extension is needed.
+    ```haskell
+    c1 :: Char
+    c1 = 'A'
+
+    c2 :: Char
+    c2 = 'b'
+    ```
+
+`Text` is the best type for most sequences of characters.  However, there's a bit of boilerplate needed to use it.
```

Comment:
I have a slight preference for not saying there's boilerplate, but just having code comments for the import line and the extension line, in the service of a "show, don't tell" style. But I won't die on this hill.